src/v0/destinations/tune/transform.js

```diff
@@ -0,0 +1,97 @@
+const get = require('get-value');
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const { defaultRequestConfig, simpleProcessRouterDest, getHashFromArray } = require('../../util');
+
+const responseBuilder = (message, { Config }) => {
+  const { tuneEvents } = Config; // Extract tuneEvents from config
+  const { properties, event: messageEvent } = message; // Destructure properties and event from message
+
+  // Find the relevant tune event based on the message's event name
+  const tuneEvent = tuneEvents.find((event) => event.eventName === messageEvent);
+
+  if (tuneEvent) {
+    const standardHashMap = getHashFromArray(tuneEvent.standardMapping);
+    const advSubIdHashMap = getHashFromArray(tuneEvent.advSubIdMapping);
+    const advUniqueIdHashMap = getHashFromArray(tuneEvent.advUniqueIdMapping);
+
+    const mapPropertiesWithNestedSupport = (msg, mappings) => {
+      const newParams = {}; // Create a new object for parameters
+      Object.entries(mappings).forEach(([key, value]) => {
+        let data; // Declare data variable
+
+        if (key.split('.').length > 1) {
+          // Handle nested keys
+          data = get(msg, key); // Use `get` to retrieve nested data
+          if (data) {
+            newParams[value] = data; // Map to the corresponding destination key
+          }
+        } else {
+          // Handle non-nested keys
+          data = get(properties, key); // Retrieve data from properties directly
```

Comment:
Need to check properties is `definedNotNull` and isNotEmpty and is a object

Reply:
I think `get` handles it but we should have a test scenario, in component test

Reply:
> Need to check properties is definedNotNull and isNotEmpty and is a object

We need to validate(isDefined & is an object) at the beginning of the transformation may be once I know this is `track` event.
Do we really need to worry about `properties` object being empty ? -- IMO don't think so. But we can validate this case

Reply:
>We need to validate(isDefined & is an object) at the beginning of the transformation may be once we know this is track event.
Do we really need to worry about properties object being empty ? -- IMO don't think so. But we can validate this case

This comment can be ignored. The intention is to make sure we move validations to the top-level rather than doing them in middle of transformation

Reply:
ok @sanpj2292